src/Cache/Context/OgMembershipStateCacheContext.php

```diff
@@ -82,25 +69,15 @@ public static function getLabel() {
    * {@inheritdoc}
    */
   public function getContext() {
-    if (!$route_contexts = $this->routeMatch->getRouteObject()->getOption('parameters')) {
-      // No "parameters" defined in the route.
-      return self::NO_CONTEXT;
-    }
-
-    if (!$entity_type_ids = array_keys($this->groupTypeManager->getAllGroupBundles())) {
-      // No group entities.
-      return self::NO_CONTEXT;
-    }
-
-    if (!$entity_type_ids = array_intersect(array_keys($route_contexts), $entity_type_ids)) {
-      // No parameters that match the group entities.
+    // Do not provide a cache context if there are no groups in the current
+    // context.
+    $contexts = $this->ogContext->getRuntimeContexts(['og']);
```

Comment:
A context provider can potentially return multiple contexts of different types, so you should pass in an array of context types that you are interested in. This allows the context provider to work more efficiently and not look up any unnecessary contexts.

Here's the relevant documentation from `ContextProviderInterface`:

```
  /**
   * Gets runtime context values for the given context IDs.
   *
   * @param string[] $unqualified_context_ids
   *   The requested context IDs. The context provider must only return contexts
   *   for those IDs.
   */
  public function getRuntimeContexts(array $unqualified_context_ids);
```